cfme/tests/webui/test_general_ui.py

```diff
@@ -567,6 +574,47 @@ def test_infrastructure_provider_left_panel_titles(
         soft_assert(test_view.is_displayed, f"{test_view} not displayed.")
 
 
+@pytest.mark.tier(1)
+@pytest.mark.provider([InfraProvider], selector=ONE_PER_CATEGORY)
+@pytest.mark.meta(
+    automates=[1733120],
+    blockers=[BZ(1733120, forced_streams=["5.10"])],
+)
+def test_compare_vm_from_datastore_relationships(appliance, setup_provider, provider):
+    """
+    Bugzilla:
+        1733120
+        1784179
+
+    Polarion:
+        assignee: pvala
+        casecomponent: Infra
+        caseimportance: medium
+        initialEstimate: 1/18h
+        setup:
+            1. Add an infra provider.
+        testSteps:
+            1. Select a datastore with at least 2 VMS, and navigate to a it's Details page.
+            2. Click on Managed VMs from the relationships table.
+            3. Select at least 2 VMs and click on `Configuration > Compare the selected items`
+        expectedResults:
+            1.
+            2.
+            3. Comparison page should be displayed, there should be no exception on the page.
+    """
+    datastore = appliance.collections.datastores.instantiate(
+        name=provider.data["provisioning"]["datastore"], provider=provider
+    )
+    view = navigate_to(datastore, "ManagedVMs")
+
+    # Check any 3 entities for comparison
+    [vm.check() for vm in sample(view.entities.get_all(), 3)]
```

Comment:
Can you use `ensure_checked` and the `slice` arg here? https://github.com/ManageIQ/integration_tests/blob/9aeff8439f638800462a284ca9855fdf807b685e/widgetastic_manageiq/__init__.py#L3456-L3463